src/dice/loader.c

```diff
@@ -39,13 +40,27 @@ _load_plugin(const char *path)
         log_fatalf("could not open %s: %s\n", path, err);
 }
 
+static char *
+_strdup(const char *str)
+{
+    if (str == NULL)
+        return NULL;
+
+    const size_t len = strlen(str) + 1;
+    char *copy       = mempool_alloc(len);
+    if (copy == NULL)
+        return copy;
+
+    return strncat(copy, str, len);
```

Comment:
why use strncat, not strncpy? is mempool_alloc calloc?

Reply:
I should have used memcpy instead. Will change it. Thx for the comment.